components/marketplace/buy-fractional-order-form.tsx

```diff
@@ -83,9 +84,10 @@ export const BuyFractionalOrderForm = ({
   const getUnitsToBuy = (percentageAmount: string) => {
     try {
       const hypercertUnits = BigInt(hypercert.units || 0);
-      const percentageAsBigInt = BigInt(Number(percentageAmount) * 100);
+      const percentageAsBigInt = parseUnits(percentageAmount, 8);
       const unitsToBuy =
-        (hypercertUnits * percentageAsBigInt) / BigInt(100 * 100);
+        (hypercertUnits * percentageAsBigInt) /
+        (BigInt(100) * DEFAULT_NUM_FRACTIONS);
```

Comment:
nit: You could write `100n`

Reply:
<img width="698" alt="afbeelding" src="https://github.com/user-attachments/assets/85e1b11c-480c-42e1-80ae-5c80d5e09d4d">
We cannot due to our ES target.